libevm/VM.cpp

```diff
@@ -186,10 +186,16 @@ void VM::interpretCases()
 
 		CASE_BEGIN(SUICIDE)
 		{
+			m_runGas = toUint64(m_schedule->suicideGas);
+			Address dest = asAddress(*m_sp);
+
+			// After EIP150 hard fork charge additional cost of sending
+			// ethers to non-existing account.
+			if (!m_schedule->staticCallDepthLimit && !m_ext->exists(dest))
```

Comment:
This requires a new bool, charging `callNewAccountGas` here is different from `staticCallDepthLimit`.
To get both readability and non-redundancy, I would suggest the following:

Add two inline functions to the schedule: `staticCallDepthLimit` and `suicideChargesNewAccountGas` and replace the `staticCallDepthLimit` by a bool member that is called `eip150Mode` and let the two inline function return the respective values depending on that member.